dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/HostFactoryImpl.java

```diff
@@ -0,0 +1,790 @@
+                dc.addParam(site.getIdentifier());
+                dc.loadResult();
+
+                Inode.Type[] assets = {Inode.Type.CONTAINERS, Inode.Type.TEMPLATE, Inode.Type.LINKS};
+                for(Inode.Type asset : assets) {
+                    dc.setSQL("select inode from "+asset.getTableName()+" where exists (select * from identifier where host_inode=? and id="+asset.getTableName()+".identifier)");
+                    dc.addParam(site.getIdentifier());
+                    for(Map row : (List<Map>)dc.loadResults()) {
+                        dc.setSQL("delete from "+asset.getVersionTableName()+" where working_inode=? or live_inode=?");
+                        dc.addParam(row.get("inode"));
+                        dc.addParam(row.get("inode"));
+                        dc.loadResult();
+
+                        dc.setSQL("delete from "+asset.getTableName()+" where inode=?");
+                        dc.addParam(row.get("inode"));
+                        dc.loadResult();
+                    }
+                }
+
+                //Remove Tags
+                APILocator.getTagAPI().deleteTagsByHostId(site.getIdentifier());
+
+                // Double-check that ALL contentlets are effectively removed
+                // before using dotConnect to kill bad identifiers
+                List<Contentlet> remainingContenlets = contentAPI
+                        .findContentletsByHost(site, user, respectFrontendRoles);
+                if (remainingContenlets != null
+                        && remainingContenlets.size() > 0) {
+                    contentAPI.deleteByHost(site, user, respectFrontendRoles);
+                }
+
+                // kill bad identifiers pointing to the host
+                dc.setSQL("delete from identifier where host_inode=?");
+                dc.addParam(site.getIdentifier());
+                dc.loadResult();
+
+                // Remove Host
+                Contentlet c = contentAPI.find(site.getInode(), user, respectFrontendRoles);
+                contentAPI.delete(c, user, respectFrontendRoles);
+
+                try {
+                    APILocator.getAppsAPI().removeSecretsForSite(site, APILocator.systemUser());
+                } catch (final Exception e) {
+                    Logger.warn(HostAPIImpl.class, String.format("An error occurred when removing secrets for site " +
+                            "'%s': %s", site, e.getMessage()), e);
+                }
+                flushAllCaches(site);
+            }
+        }
+        final DeleteHostThread deleteHostThread = new DeleteHostThread();
+
+        if(runAsSeparatedThread) {
+            final DotConcurrentFactory concurrentFactory = DotConcurrentFactory.getInstance();
+            future = Optional.of(concurrentFactory.getSubmitter
+                    (DotConcurrentFactory.DOT_SYSTEM_THREAD_POOL).submit(deleteHostThread));
+        } else {
+            deleteHostThread.call();
```

Comment:
you can use this
```
future = ConcurrentUtils.constantFuture(deleteHostThread.call());
```

In order to get a Future with the value returned on the sync method